pkg/api/vm/handler.go

```diff
@@ -593,6 +617,81 @@ func (h *vmActionHandler) updateVMVolumeClaimTemplate(vm *kubevirtv1.VirtualMach
 	return nil
 }
 
+func (h *vmActionHandler) getPVCStorageClassMap(vm *kubevirtv1.VirtualMachine) (map[string]string, error) {
+	pvcStorageClassMap := map[string]string{}
+	for _, volume := range vm.Spec.Template.Spec.Volumes {
+		if volume.PersistentVolumeClaim == nil {
+			continue
+		}
+
+		pvc, err := h.pvcCache.Get(vm.Namespace, volume.PersistentVolumeClaim.ClaimName)
+		if err != nil {
+			return pvcStorageClassMap, err
+		}
+		storageClassName := settings.DefaultStorageClass.Get()
+		if imageID, ok := pvc.Annotations[util.AnnotationImageID]; ok {
+			imageIDSplit := strings.Split(imageID, "/")
+			if len(imageIDSplit) == 2 {
+				vmImage, err := h.vmImageCache.Get(imageIDSplit[0], imageIDSplit[1])
+				if err != nil {
+					return pvcStorageClassMap, err
+				}
+				if _, ok = vmImage.Annotations[util.AnnotationStorageClassName]; ok {
+					storageClassName = vmImage.Annotations[util.AnnotationStorageClassName]
+				}
+			}
+		}
+
+		sc, err := h.storageClassCache.Get(storageClassName)
+		if err != nil {
+			return pvcStorageClassMap, err
+		}
+		if sc.Provisioner != "driver.longhorn.io" {
```

Comment:
s/driver.longhorn.io/longhorntypes.LonghornDriverName/

Reply:
Updated it. Thanks.